src/node.cc

```diff
@@ -1110,6 +1111,57 @@ bool ShouldAbortOnUncaughtException(Isolate* isolate) {
 }
 
 
+void PromiseHook(PromiseHookType type,
+                 Local<Promise> promise,
+                 Local<Value> parent) {
+  Environment* env = Environment::GetCurrent(Isolate::GetCurrent());
+  Local<Context> context = env->context();
+
+  if (type == PromiseHookType::kInit && env->in_domain()) {
```

Comment:
I’m pretty sure, yes. This is based on existing code for similar circumstances (plain JS calls coming from the void), and `in_domain()` will automatically be true between the enter and exit callbacks.